src/plugins/es_ui_shared/static/forms/hook_form_lib/components/use_field.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect } from 'react';
+
+import { Form, Field as FieldType, FieldConfig } from '../types';
+import { useField } from '../hooks';
+
+interface Props {
+  path: string;
+  config?: FieldConfig<any>;
+  defaultValue?: unknown;
+  form: Form<any>;
+  component?: (({ field }: { field: FieldType } & any) => JSX.Element) | 'input';
+  componentProps?: any;
+  children?: (field: FieldType) => JSX.Element;
+}
+
+export const UseField = ({
```

Comment:
I was just wondering, for these re-usable form building blocks. Would it be worth exploring how we could reduce the number of required props (mainly thinking of `form`) by using `React.createContext`? Since the intention is that they will _always_ by used as a child of `useForm`'s `form` component.

Reply:
Or, actually, I see now that is not how `use_form.ts` is used. It's slightly confusing that `form`, from `use_form.ts` and `form`, the JSX element have the same names. If we were to explore my suggestion you would need to create your own `Form` element that encapsulates react context and acts as the provider.

Reply:
I like the recurring pattern of a `<UseForm>` root element that surrounds `<UseField>` children (I believe this is the pattern you're proposing, correct me if I'm wrong).

Reply:
Great idea @jloleysens ! I will add a `<FormProvider form={form}>` that must exist around the form and remove the need of drilling the `form` prop all around.

Reply:
@cjcenizal Yeah so using `UseField` elements inside of some wrapping, context providing component 😊